### PR TITLE
Fix Typescript errors

### PR DESCRIPTION
**Pull request by guillaumevincent**
I tried by accident to use Typescript 4.8. And ts-linter raised some issues.
This patch fixes most of the issues I found.

### CODE REVIEW

**Comment by guillaumevincent** [comment]
Here https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/ocm/components/clusters/ClusterPage.tsx#L66 resetCluster  expects a Promise. And forceReload is a redux action. I don't understand how it's working because we never dispatch the action. I bet it's not working.

I transformed the function into an async function

```
  const forceReloadAsync = () =>
    new Promise<void>((resolve) => {
      dispatch(forceReload());
      resolve();
    });
```

But maybe we can transform resetCluster into a basic function instead of a promise.

**Comment by celdrake** [comment]
> Here https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/ocm/components/clusters/ClusterPage.tsx#L66 resetCluster expects a Promise. And forceReload is a redux action. I don't understand how it's working because we never dispatch the action. I bet it's not working.
> 
> I transformed the function into an async function
> 
> ```
>   const forceReloadAsync = () =>
>     new Promise<void>((resolve) => {
>       dispatch(forceReload());
>       resolve();
>     });
> ```
> 
> But maybe we can transform resetCluster into a basic function instead of a promise.

It's weird, as you say I think it must not have worked until now.
This would imply a few changes in `use-hosts-table` + testing so maybe for now let's keep it with your changes.

File: src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const ACMFeatureSupportLevelProvider: React.FC<ACMFeatureSupportLevelProv
   const getDisabledReasonCallback = React.useCallback(
     (versionName: string, featureId: FeatureId) => {
       const isSupported = isFeatureSupportedCallback(versionName, featureId);
-      return getFeatureDisabledReason(featureId, undefined, isSupported);
+      return getFeatureDisabledReason(featureId, undefined, isSupported, (s: string) => s);
```
Comment (rawagner) [inline src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx:96]: you should pass `t` from `i18n`

```suggestion
      return getFeatureDisabledReason(featureId, undefined, isSupported, t);
```

```ts

import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
...
const { t } = useTranslation();
...
return getFeatureDisabledReason(featureId, undefined, isSupported, t);
```

**Review by rawagner — CHANGES_REQUESTED** [review]

File: src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx (hunk 1 above)
Comment (guillaumevincent) [reply src/cim/components/featureSupportLevels/ACMFeatureSupportLevelProvider.tsx]: done thank you

File: src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx [hunk 2]
```diff
@@ -82,7 +82,6 @@ const AdvancedNetworkFields = ({ isSDNSelectable }: AdvancedNetworkFieldsProps)
                     label={`Cluster network CIDR${networkSuffix}`}
                     helperText={clusterCidrHelperText}
                     isRequired
-                    labelInfo={index === 0 && isDualStack ? 'Primary' : ''}
```
[the remark this answers is not in the document]
Comment (guillaumevincent) [reply src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx]: I rollback my change, I don't fully understand why the linter think OcmInputField is InputFieldProps only

**Comment by guillaumevincent** [comment]
Please take special attention to this pull request. I have not tested live. I'm not very confident with this resetCluster change. If it was working before, then we should find a way to test the new behavior.

I'm also okay to put this PR on hold or draft if needed.

File: src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (celdrake) [reply src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx]: The problem is here: https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/ocm/components/ui/OcmFormFields.tsx#L49
The proper definition should match that of `InputField`

**Comment by celdrake** [comment]
@guillaumevincent I suggest we split the issue about `resetCluster` until we can test it. Otherwise it may take a long time to merge

File: src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (celdrake) [reply src/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx]: It's been fixed already in `master`

**Comment by guillaumevincent** [comment]
closing for now, create more noise than help

**Pull request closed without merging**